electroncash_plugins/satochip/qt.py

```diff
@@ -48,9 +52,16 @@ def settings_dialog(self, window):
 
     def show_settings_dialog(self, window, keystore):
         # When they click on the icon for Satochip we come here.
-        device_id = self.choose_device(window, keystore)
-        if device_id:
-            SatochipSettingsDialog(window, self, keystore, device_id).exec_()
+        # device_id = self.choose_device(window, keystore)
+        # if device_id:
+            # SatochipSettingsDialog(window, self, keystore, device_id).exec_()
+        def connect():
+            device_id = self.choose_device(window, keystore)
+            return device_id
+        def show_dialog(device_id):
+            if device_id:
+                SatochipSettingsDialog(window, self, keystore, device_id).exec_()
+        keystore.thread.add(connect, on_success=show_dialog)
```

Comment:
Wow this is nuts. Yeah ok so this throws up a UI in the main thread and waits for a response, then on success shows the settings dialog?  Ok.. this threw me for a loop initially but it seems correct..